### DISCUSSION TITLE
Missing Video End Status Feedback in expo-video for Splash Screen

### DISCUSSION THREAD

**Discussion by vitorGabr** [expo/expo#33651] [Archived: Questions] [answered]
I noticed that the expo-video library is recommended over expo-av for video playback. However, I couldn't find any implementation of a status or feedback to detect when a video finishes playing.

This feature is particularly important for use cases like implementing a splash screen where actions need to be triggered upon the video's end.

Could you confirm if this feature exists or if there are plans to implement it in the future? If not, could you suggest a workaround to achieve this functionality using expo-video?

**Comment by yogiw** [comment]
I ended up listening to the current time and comparing it with the video's duration, something like this:

```tsx
 const { isPlaying } = useEvent(player, "playingChange", {
    isPlaying: player.playing,
  })

  useEffect(() => {
    const finishedThreshold = player.duration * FINISHED_THRESHOLD
    if (player.currentTime >= finishedThreshold && player.currentTime > 0) {
      props.onEnd?.()
    }
  }, [player.currentTime, isPlaying])
```

**Reply by tsapeta** [reply]
I think the above could be a bit simpler and more efficient with `useEventListener` instead of `useEvent` (which creates a new state). Anyway, see my comment below for the best solution.

**Reply by gitstashapply** [reply]
I ended up with this:
```
    useEffect(() => {
      const statusSubscription = player?.addListener(
        'playingChange',
        ({isPlaying}) => {
          const finishedThreshold = player.duration * 0.99;
          if (
            player.currentTime >= finishedThreshold &&
            player.currentTime > 0
          ) {
            setPlayerStatus(VideoState.ENDED);
            return;
          }

          setPlayerStatus(isPlaying ? VideoState.PLAYING : VideoState.PAUSED);
        },
      );

      return () => {
        statusSubscription.remove();
      };
    }, []);
    
```
    
    REMARK: 
    1. I needed reactive state for specific reasons
    2. I tried to avoid 'race condition' while using `playToEnd` event (both `playingChange` and `playToEnd` was called almost simultaneously)

**Comment by tsapeta** [comment, marked as answer]
Please check out `playToEnd` event, it's supposed to do what you want 😉

**Comment by SyntheticTV** [comment]
Thank you tsapeta. I was stuck with this same issue. This worked for me! I didn't even see it on the expo-video page until you pointed it out. 

  useEventListener(player, 'playToEnd', () => {
    console.log(`${Platform.OS}: Video reached end naturally`);
    playNextVideo();
  });
